ragstack/colbertbase/colbertbase/astra_db.py

```diff
@@ -0,0 +1,177 @@
+        ORDER BY bert_embedding ANN OF ?
+        LIMIT ?
+        """
+        self.query_colbert_ann_stmt = self.session.prepare(query_colbert_ann_cql)
+
+        query_colbert_parts_cql = f"""
+        SELECT title, part, bert_embedding
+        FROM {keyspace}.colbert_embeddings
+        WHERE title = ? AND part = ?
+        """
+        self.query_colbert_parts_stmt = self.session.prepare(query_colbert_parts_cql)
+
+        query_part_by_pk = f"""
+        SELECT body
+        FROM {keyspace}.chunks
+        WHERE title = ? AND part = ?
+        """
+        self.query_part_by_pk_stmt = self.session.prepare(query_part_by_pk)
+
+        print("statements are prepared")
+
+    def create_tables(self):
+        self.session.execute(f"""
+            use {self.keyspace};
+        """)
+        print(f"Using keyspace {self.keyspace}")
+
+        self.session.execute("""
+            CREATE TABLE IF NOT EXISTS chunks(
+                title text,
+                part int,
+                body text,
+                PRIMARY KEY (title, part)
+            );
+        """)
+        print("Created chunks table")
+
+        self.session.execute("""
+            CREATE TABLE IF NOT EXISTS colbert_embeddings (
+                title text,
+                part int,
+                embedding_id int,
+                bert_embedding vector<float, 128>,
+                PRIMARY KEY (title, part, embedding_id)
+            );
+        """)
+        print("Created colbert_embeddings table")
+
+        self.create_index("""
+            CREATE CUSTOM INDEX colbert_ann ON colbert_embeddings(bert_embedding) USING 'StorageAttachedIndex'
+  WITH OPTIONS = { 'similarity_function': 'DOT_PRODUCT' };
+        """)
+        print("Created index on colbert_embeddings table")
+                             
+    def create_index(self, command: str):
+        try:
+            self.session.execute(command)
+        except InvalidRequest as e:
+            if "already exists" in str(e):
+                print("Index already exists and continue...")
+            else:
+                raise e
+        # throw other exceptions
+
+    # ensure db connection is alive
+    def ping(self):
+        self.session.execute("select release_version from system.local").one()
+
+
+    def insert_chunk(self, title: str, part: int, body: str):
+        self.session.execute(self.insert_chunk_stmt, (title, part, body))
+    
+    def insert_colbert_embeddings_chunks(
+        self,
+        embeddings: List[PassageEmbeddings],
+        delete_existed_passage: bool = False
+    ) -> None:
+        if delete_existed_passage:
+            for p in embeddings:
+                try:
+                    self.delete_title(p.title())
+                except Exception as e:
+                    # no need to throw error if the title does not exist
+                    # let the error propagate
+                    print(f"delete title {p.title()} error {e}")
```

Comment:
> no need to throw error if the title does not exist

if the title doens't exists, you shouldn't have any error from the DELETE statement. We should instead raise error if something is not working in the session